smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,688 @@
+            [pattern], ignore_patterns, ignore_directories, case_sensitive
+        )  # type: ignore
+        self._logger = logger
+        self._tracked_runs: t.Dict[int, Run] = {}
+        self._tracked_jobs: t.Dict[_JobKey, JobEntity] = {}
+        self._completed_jobs: t.Dict[_JobKey, JobEntity] = {}
+        self._launcher_type: str = ""
+        self._launcher: t.Optional[Launcher] = None
+        self.job_manager: JobManager = JobManager(threading.RLock())
+        self._launcher_map: t.Dict[str, t.Type[Launcher]] = {
+            "slurm": SlurmLauncher,
+            "pbs": PBSLauncher,
+            "cobalt": CobaltLauncher,
+            "lsf": LSFLauncher,
+            "local": LocalLauncher,
+        }
+
+    def init_launcher(self, launcher: str) -> Launcher:
+        """Initialize the controller with a specific type of launcher.
+        SmartSim currently supports slurm, pbs(pro), cobalt, lsf,
+        and local launching
+
+        :param launcher: which launcher to initialize
+        :type launcher: str
+        :raises SSUnsupportedError: if a string is passed that is not
+                                    a supported launcher
+        :raises TypeError: if no launcher argument is provided.
+        """
+        if not launcher:
+            raise TypeError("Must provide a 'launcher' argument")
+
+        if launcher_type := self._launcher_map.get(launcher.lower(), None):
+            return launcher_type()
+
+        raise ValueError("Launcher type not supported: " + launcher)
+
+
+    def set_launcher(self, launcher_type: str) -> None:
+        """Set the launcher for the experiment"""
+        if launcher_type != self._launcher_type:
+            self._launcher_type = launcher_type
+            self._launcher = self.init_launcher(launcher_type)
+            self.job_manager.set_launcher(self._launcher)
+            self.job_manager.add_job_onstep_callback(track_timestep)
+
+            self.job_manager.start()
+
+    @property
+    def launcher(self) -> Launcher:
+        """Return a launcher appropriate for the experiment"""
+        if not self._launcher:
+            self._launcher = LocalLauncher()
+
+        return self._launcher
+
+    def process_manifest(self, manifest_path: str) -> None:
+        """Read the runtime manifest for the experiment and track new entities
+
+        :param manifest_path: The full path to the manifest file
+        :type manifest_path: str"""
+        try:
+            manifest = load_manifest(manifest_path)
+            if not manifest:
+                return
+        except json.JSONDecodeError:
+            self._logger.error(f"Malformed manifest encountered: {manifest_path}")
+            return
+        except ValueError:
+            self._logger.error("Manifest content error", exc_info=True)
+            return
+
+        self.set_launcher(manifest.launcher)
+
+        if not self._launcher:
+            raise SmartSimError(f"Unable to set launcher from {manifest_path}")
```

Comment:
Not wrong, but theoretically un-reachable code block here. `set_launcher` -> `self._launcher = self.init_launcher(...)` and `self.init_launcher(...)` will always return a non-nullish value.

Willing to keep this block around for safety though

Reply:
removed